Base/usr/share/man/man1/grep.md

```diff
@@ -1,31 +1,56 @@
 ## Name
 
-grep
+grep - Find lines in files
 
 ## Synopsis
 
-```sh
-$ grep [--recursive] [--extended-regexp] [--regexp Pattern] [-i] [--line-numbers] [--invert-match] [--quiet] [--no-messages] [--binary-mode ] [--text] [-I] [--color WHEN] [--count] [file...]
+```**sh
+$ grep [options...] <pattern> [files...]
 ```
 
-## Options:
+## Description
+
+`grep` searches in `files` (or standard input) for lines containing `pattern`.
+`pattern` is a POSIX basic regular expression. For extended regular expressions, use `-E`.
+
+## Options
 
 * `-r`, `--recursive`: Recursively scan files
-* `-E`, `--extended-regexp`: Extended regular expressions
-* `-e Pattern`, `--regexp Pattern`: Pattern
+* `-E`, `--extended-regexp`: Use extended regular expressions
+* `-e pattern`, `--regexp pattern`: pattern
 * `-i`: Make matches case-insensitive
 * `-n`, `--line-numbers`: Output line-numbers
 * `-v`, `--invert-match`: Select non-matching lines
 * `-q`, `--quiet`: Do not write anything to standard output
 * `-s`, `--no-messages`: Suppress error messages for nonexistent or unreadable files
-* `--binary-mode`: Action to take for binary files ([binary], text, skip)
-* `-a`, `--text`: Treat binary files as text (same as --binary-mode text)
-* `-I`: Ignore binary files (same as --binary-mode skip)
-* `--color WHEN`: When to use colored output for the matching text ([auto], never, always)
+* `--binary-mode`: Action to take for binary files: binary (default), text, skip
+* `-a`, `--text`: Treat binary files as text (same as `--binary-mode text`)
+* `-I`: Ignore binary files (same as `--binary-mode skip`)
+* `--color WHEN`: When to use colored output for the matching text: auto (default), never, always
```

Comment:
If you want to fix the format, why not fix the `ArgParser` generator so it generates them in the desired format?

Reply:
Sure thing, but that's out of scope for this PR